limit size of process and jobs dictionary

In `core/process.py JobState`, we use a dictionary to map PID -> `Process` object, etc. for `waitpid()`.

This dictionary is currently unlimited in size, which should probably be considered a memory leak

In `process.py`, we find the following (starting at line 1343)

```
  def MaybeRemove(self, pid):                                                                                                                                  
    # type: (int) -> None                                                                                                                                      
    """Process and Pipeline can call this."""                                                                                                                  
    # Problem: This only happens after an explicit wait()?                                                                                                     
    # I think the main_loop in bash waits without blocking?                                                                                                    
    log('JobState MaybeRemove %d', pid)                                                                                                                        
                                                                                                                                                               
    # TODO: Enabling this causes a failure in spec/background.                                                                                                 
    if 0:                                                                                                                                                      
      try:                                                                                                                                                     
        del self.jobs[pid]                                                                                                                                     
      except KeyError:                                                                                                                                         
        # This should never happen?                                                                                                                            
        log("AssertionError: PID %d should have never been in the job list", pid)                                                                              
```

Wondering if you can provide any insight into why the spec tests were failing.

--- a/core/process.py
+++ b/core/process.py
@@ -1154,7 +1154,8 @@ class Process(Job):
                     print_stderr('[%d] Done PID %d' % (self.job_id, self.pid))
 
                 self.job_list.RemoveJob(self.job_id)
-                self.job_list.RemoveChildProcess(self.pid)
+
+            self.job_list.RemoveChildProcess(self.pid)
 
             if not self.in_background:
                 self.job_control.MaybeTakeTerminal()

